1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -10,16 +53,17 @@ function setup() {
   });
 
   document.getElementById("stop").addEventListener("click", () => {
-    pauseAlarm();
+    stopAlarm();
   });
 }
 
 function playAlarm() {
   audio.play();
 }
 
-function pauseAlarm() {
+function stopAlarm() {
   audio.pause();
+  clearInterval(bgState);
```

Comment:
Stopping the alarm doesn't always work - I believe this is because you are not clearing the alarm timer when pressing the stop button. Sometimes the clock time displayed goes negative - I'd suggest revisiting the code and clearing all the timers that you have set when pressing the stop button and pausing buttons.